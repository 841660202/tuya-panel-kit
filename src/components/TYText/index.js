import React from 'react';
// import PropTypes from 'prop-types';
import { Text } from 'react-native';

class TYText extends React.Component {
  static propTypes = {
    ...Text.propTypes,
  };
  constructor(props) {
    super(props);
    this.state = {
      text: props.text ? props.text : props.children
    };
  }
  componentWillReceiveProps(nextProps) {
    this.setState({ text: nextProps.text ? nextProps.text : nextProps.children });
  }
  setText = text => {
    this.setState({ text });
  }
  setNativeProps = nativeProps => {
    this._text.setNativeProps(nativeProps);
  }
  render() {
    const { style, ...props } = this.props;
    return (
      <Text
        ref={ref => { this._text = ref; }}
        allowFontScaling={false}
        style={[{ color: '#fff', fontSize: 16 }, style]}
        {...props}
      >
        { this.state.text }
      </Text>
    );
  }
}
export default TYText;
