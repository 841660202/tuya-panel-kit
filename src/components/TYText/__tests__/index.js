import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TYText from '../index';

describe('TYText Component', () => {
  it('basic render', () => {
    const component = renderer.create(<TYText/>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
