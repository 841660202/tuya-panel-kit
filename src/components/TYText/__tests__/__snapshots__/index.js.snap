// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TYText Component basic render 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fff",
        "fontSize": 16,
      },
      undefined,
    ]
  }
/>
`;
